From bcfeea5efa167ea711a84c94cab8e341d8fac975 Mon Sep 17 00:00:00 2001
From: Isaac Chen <ycchen@google.com>
Date: Tue, 12 Sep 2017 16:47:54 +0800
Subject: [PATCH 11/23] Enable 32-bit armv8 NEON for libm

Since ARM neon instructions were only used on armv7-a-neon architecture
variant, the default implementation for 32-bit armv8-a cores doesn't
use these advanced SIMD instructions. By using "neon" key in the
Android.bp, both arch variants (armv7-a-neon and armv8-a) are covered.

Bug: 65569003
Test: lunch aosp_arm64; emulator # on oc-mr1-dev; boot to home screen
      Also checked if sqrt.o and floor.o are actually built and linked
      in the resulted libm.a for both armv7-a-neon/cortex-a15 and
      armv8-a/generic (2nd) arch/cpu variant.
Change-Id: I2084dbdb12e252b06bba5adc65adb59e97a99332
---
 libm/Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libm/Android.bp b/libm/Android.bp
index 0bdab940a..1d8edd7b6 100644
--- a/libm/Android.bp
+++ b/libm/Android.bp
@@ -285,7 +285,7 @@ cc_library {
             srcs: [
                 "arm/fenv.c",
             ],
-            armv7_a_neon: {
+            neon: {
                 srcs: [
                     "arm/sqrt.S",
                     "arm/floor.S",
-- 
2.11.0

